### PR TITLE
Remove files related to MBED OS (not mbedtls!)

### PR DESCRIPTION
**Pull request by ewertons**
MBED OS support has not been maintained for quite a while (years).
We are removing support for MBED OS.
Support for MBED TLS will still be kept.

### CODE REVIEW

File: build/.vsts-ci.yml [hunk 1]
```diff
@@ -79,6 +79,8 @@ jobs:
       IOTHUB_CONN_STRING_INVALIDCERT: $(IOTHUB-CONN-STRING-INVALIDCERT)
       DPS_GLOBALDEVICEENDPOINT_INVALIDCERT: $(DPS-GLOBALDEVICEENDPOINT-INVALIDCERT)
       PROVISIONING_CONNECTION_STRING_INVALIDCERT: $(PROVISIONING-CONNECTION-STRING-INVALIDCERT)
+  - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
+    displayName: 'Component Detection'
   - script: cd .. && rd /Q /S $(Agent.BuildDirectory)\s
```
Comment (ericwol-msft) [inline build/.vsts-ci.yml:84]: we don't need to clean up build with 1ES images. The trashcan still needs the cleanup.

**Review by ericwol-msft — APPROVED** [review]

File: build/.vsts-ci.yml (hunk 1 above)
Comment (ewertons) [reply build/.vsts-ci.yml:84]: Cool, I'm updating the other jobs to follow that

File: build/.vsts-ci.yml [hunk 2]
```diff
@@ -662,7 +612,3 @@ jobs:
      chmod +x jenkins/raspberrypi_c_buster.sh
      ./jenkins/raspberrypi_c_buster.sh
     displayName: 'Build'
-  - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
```
Comment (ericwol-msft) [inline build/.vsts-ci.yml]: OSX still needs cleanup

**Review by ericwol-msft — APPROVED** [review]

**Review by danewalton — APPROVED** [review]

File: build/.vsts-ci.yml (hunk 1 above)
Comment (vaavva) [reply build/.vsts-ci.yml:84]: Should be just OSX and XCode Native that need the cleanup step!

Comment (ewertons) [reply build/.vsts-ci.yml:84]: Done.

Comment (vaavva) [reply build/.vsts-ci.yml:84]: Cleanup steps for windowsdynamic, windowsx64, and windowsx86 should be removed as well

**Pull request merged**